x-pack/test/performance/scalability/config.ts

```diff
@@ -0,0 +1,108 @@
+const APM_SERVER_URL = 'https://142fea2d3047486e925eb8b223559cae.apm.europe-west1.gcp.cloud.es.io';
+const APM_PUBLIC_TOKEN = 'pWFFEym07AKBBhUE2i';
+const AGGS_SHARD_DELAY = process.env.LOAD_TESTING_SHARD_DELAY;
+const DISABLE_PLUGINS = process.env.LOAD_TESTING_DISABLE_PLUGINS;
+const scalabilityJsonPath = process.env.SCALABILITY_JOURNEY_PATH;
+const gatlingProjectRootPath: string =
+  process.env.GATLING_PROJECT_PATH || path.resolve(REPO_ROOT, '../kibana-load-testing');
+
+const readScalabilityJourney = (filePath: string): ScalabilityJourney => {
+  if (path.extname(filePath) !== '.json') {
+    throw createFlagError(`Path to scalability journey json is non-json file: '${filePath}'`);
+  }
+  try {
+    return JSON.parse(fs.readFileSync(filePath, 'utf-8'));
+  } catch (error) {
+    if (error.code === 'ENOENT') {
+      throw createFlagError(`Path to scalability journey json is invalid: ${filePath}`);
+    }
+    throw createFlagError(`Invalid JSON provided: '${filePath}', ${error}`);
+  }
+};
+
+interface ScalabilityJourney {
+  journeyName: string;
+}
+
+export default async function ({ readConfigFile }: FtrConfigProviderContext) {
+  const performanceConfig = await readConfigFile(require.resolve('../journeys/base.config.ts'));
+  const functionalConfig = await readConfigFile(require.resolve('../../functional/config.base.js'));
+
+  if (!fs.existsSync(gatlingProjectRootPath)) {
+    throw createFlagError(
+      `Incorrect path to load testing project: '${gatlingProjectRootPath}'\n
+    Clone 'elastic/kibana-load-testing' and set path using 'GATLING_PROJECT_PATH' env var`
+    );
+  }
+
+  if (!scalabilityJsonPath) {
+    throw createFlagError(
+      `Set path to scalability journey json using 'SCALABILITY_JOURNEY_PATH' env var`
+    );
+  }
+  const scalabilityJourney = readScalabilityJourney(scalabilityJsonPath);
+
+  const apmGlobalLabels = {
+    ...performanceConfig.get('kbnTestServer').env.ELASTIC_APM_GLOBAL_LABELS,
+    journeyFilePath: path.basename(scalabilityJsonPath),
+    journeyName: scalabilityJourney.journeyName,
+  };
+
+  return {
+    ...performanceConfig.getAll(),
+
+    testRunner: (context: FtrProviderContext) =>
+      ScalabilityTestRunner(context, scalabilityJsonPath, gatlingProjectRootPath),
+
+    esTestCluster: {
+      ...performanceConfig.get('esTestCluster'),
+      serverArgs: [...performanceConfig.get('esTestCluster.serverArgs')],
+      esJavaOpts: '-Xms8g -Xmx8g',
+    },
+
+    kbnTestServer: {
+      ...performanceConfig.get('kbnTestServer'),
+      sourceArgs: [
+        ...performanceConfig.get('kbnTestServer.sourceArgs'),
+        '--no-base-path',
+        '--env.name=development',
+        ...(!!AGGS_SHARD_DELAY ? ['--data.search.aggs.shardDelay.enabled=true'] : []),
+        ...(!!DISABLE_PLUGINS ? ['--plugins.initialize=false'] : []),
+      ],
+      // Telemetry is temporary off for scalability testing
+      serverArgs: [...functionalConfig.get('kbnTestServer.serverArgs')],
```

Comment:
I was trying to re-use journeys `base.config.ts` to start Kibana with the same properties single user perf pipeline does (APM configuration is a bit different though to make it work with high load), however 07aa6d9586133c9ff4dcdaa3b72306089ec3e7d0 breaks scalability pipeline with `Error: [config validation of [telemetry].labels.ciBuildName]: definition for this key is missing`

I was not able to find a root cause in a reasonable time, so I think we can move on with running scalability testing with disabled telemetry. Fix can be done in a separate PR.
 Thoughts @suchcodemuchwow @spalger @danielmitterdorfer ?

Reply:
Spencer helped me understand, I was using not latest Kibana build. I will push a change a re-run both pipelines